jsettlers.common/src/main/java/jsettlers/common/buildings/EBuildingType.java

```diff
@@ -169,46 +158,30 @@
 		requestStacks = file.getRequestStacks();
 		offerStacks = file.getOfferStacks();
 
-		workradius = file.getWorkradius();
-		workcenter = file.getWorkcenter();
+		workRadius = file.getWorkradius();
+		workCenter = file.getWorkcenter();
+		mine = file.isMine();
 		flag = file.getFlag();
 		bricklayers = file.getBricklayers();
-		occupyerPlaces = file.getOccupyerPlaces();
+		occupierPlaces = file.getOccupyerPlaces();
 		guiImage = file.getGuiImage();
-		ImageLink[] tempimages = file.getImages();
-		if (tempimages.length == 0) {
-			// TODO: this can be removed if all images are converted
-			System.out.println("WARNING: Building " + this.toString() + " does not have an image definition.");
-			images = new ImageLink[] { new OriginalImageLink(EImageLinkType.SETTLER, IMAGE_ROMAN, imageIndex, 0) };
-			buildImages = new ImageLink[] { new OriginalImageLink(EImageLinkType.SETTLER, IMAGE_ROMAN, imageIndex, 1) };
-		} else {
-			images = tempimages;
-			buildImages = file.getBuildImages();
-		}
 
-		buildmarks = file.getBuildmarks();
+		images = file.getImages();
+		buildImages = file.getBuildImages();
+
+		buildMarks = file.getBuildmarks();
 		groundTypes = EnumSet.copyOf(file.getGroundtypes());
-		viewdistance = file.getViewdistance();
+		viewDistance = file.getViewdistance();
 
 		this.numberOfConstructionMaterials = calculateNumberOfConstructionMaterials();
 
 		this.buildingAreaBitSet = new BuildingAreaBitSet(getBuildingArea());
-	}
 
-	/**
-	 * Converts the list of supported ground types to a long for fast lookup.
-	 * 
-	 * @param groundtypeArray
-	 *            The array of ground types.
-	 * @return A packed bitset with each bit set if the ground type with that ordinal is in it.
-	 */
-	private long packGroundtypes(ELandscapeType[] groundtypeArray) {
-		assert ELandscapeType.VALUES.length <= SIZEOF_LONG : "Too many ground types.";
-		long res = 0;
-		for (ELandscapeType g : groundtypeArray) {
-			res |= (1 << g.ordinal);
+		if (mine) {
+			int centerIndex = -buildingAreaBitSet.minX - buildingAreaBitSet.minY * buildingAreaBitSet.width;
```

Comment:
This should be moved to the BuildingAreaBitSet class. Something like setCenter(1, 1) ?